tests/test_parameters_clarity.py

```diff
@@ -0,0 +1,170 @@
+        clarity(edc, early_time_limit=-1)
+    
+    # Test time limit beyond signal length
+    with pytest.raises(ValueError, match=re.escape(expected_error_message)):
+        clarity(edc, early_time_limit=200000)
+
+def test_clarity_preserves_multichannel_shape():
+    energy = np.ones((2,2,10)) / (1+np.arange(10))
+    edc = make_edc_from_energy(energy, 10)
+    output = clarity(edc, early_time_limit=80)
+    assert edc.cshape == output.shape
+
+
+def test_clarity_returns_nan_for_zero_signal():
+    edc = pf.TimeData(np.zeros((1, 128)), np.arange(128) / 1000)
+    result = clarity(edc)
+    assert np.isnan(result)
+
+
+def test_clarity_warns_for_unusually_short_early_time_limit():
+    energy = np.zeros(128)
+    edc = make_edc_from_energy(energy, sampling_rate=44100)
+    early_time_limit = 0.05
+    expected_warning_message = (
+        f"early_time_limit={early_time_limit} ms is unusual."
+        "According to DIN EN ISO 3382-3, typically 50 ms (C50) or 80 ms (C80) are chosen."
+    )
+    
+    with pytest.warns(UserWarning, match=re.escape(expected_warning_message)):
+        clarity(edc, early_time_limit)
+
+def test_clarity_calculates_known_reference_value():
+    # Linear decay → early_time_limit at 1/2 energy -> ratio = 1 -> 0 dB
+    edc_vals = np.array([1.0, 0.75, 0.5, 0.0])  # monotonic decay
+    times = np.arange(len(edc_vals)) / 1000
+    edc = pf.TimeData(edc_vals[np.newaxis, :], times)
+
+    result = clarity(edc, early_time_limit=2)
+    np.testing.assert_allclose(result, 0.0, atol=1e-6)
+
+
+def test_clarity_matches_analytical_geometric_decay_solution():
+    sampling_rate = 1000
+    decay_factor = 0.9
+    total_samples = 200
+    early_cutoff = 80  # ms
+
+    time_axis = np.arange(total_samples)
+    energy = decay_factor ** (2 * time_axis)  # squared amplitude
+    edc = make_edc_from_energy(energy, sampling_rate=sampling_rate)
+
+    squared_factor = decay_factor ** 2
+    early_energy = (1 - squared_factor ** early_cutoff) / (1 - squared_factor)
+    late_energy = (
+        squared_factor**early_cutoff - squared_factor**total_samples
+    ) / (1 - squared_factor)
+    expected_db = 10 * np.log10(early_energy / late_energy)
+
+    result = clarity(edc, early_time_limit=early_cutoff)
+    np.testing.assert_allclose(result, expected_db, atol=1e-6)
+
+def test_clarity_values_for_given_ratio():
+    energy_early = 1
+    energy_late = .5
+    etc = pf.TimeData(np.zeros((3, 1000)), np.arange(1000) / 1000)
+    etc.time[..., 10] = energy_early
+    etc.time[..., 100] = energy_late
+    edc = ra.edc.schroeder_integration(etc, is_energy=True)
+    edc = pf.dsp.normalize(edc, reference_method='max')
+    result = clarity(edc, early_time_limit=80)
+    clarity_value_db = 10 * np.log10(energy_early/energy_late)
+    npt.assert_allclose(result, clarity_value_db, atol=1e-6)
+
+def test_clarity_from_truth_edc():
+    # real-EDC from test_edc:test_edc_eyring
+    real_etc = np.array([
```

Comment:
```suggestion
    real_edc = np.array([
```
because etc is often used for energy time curve, e.g. p^2(t)